api/src/main/java/marquez/service/DatasetService.java

```diff
@@ -265,6 +277,26 @@ public boolean fieldExists(
     }
   }
 
+  public Optional<DatasetVersion> getVersion(@NonNull Version version) {
+    return datasetVersionDao.findBy(version.getValue()).map(this::toDatasetVersion);
+  }
+
+  public ImmutableList<DatasetVersion> getVersionsFor(
```

Comment:
I think an easier semantic to maintain would be that everything is assumed to be immutable unless explicitly labeled as mutable for semantic reasons (e.g. getMutableVersionsFor..). Tying to Guava ImmutableList is a bit awkward since Java also has immutable lists.

Reply:
Agreed. Let's move towards having our services, models, etc use interfaces (in this case a `Map`) and not have a type defined in the guava lib as a return type.